llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1092,6 +1024,86 @@ void VPlanTransforms::simplifyRecipes(VPlan &Plan, Type &CanonicalIVTy) {
   }
 }
 
+/// Normalize and simplify VPBlendRecipes. Should be run after simplifyRecipes
+/// to make sure the masks are simplified.
+static void simplifyBlends(VPlan &Plan) {
+  using namespace llvm::VPlanPatternMatch;
+  ReversePostOrderTraversal<VPBlockDeepTraversalWrapper<VPBlockBase *>> RPOT(
+      Plan.getEntry());
+  for (VPBasicBlock *VPBB : VPBlockUtils::blocksOnly<VPBasicBlock>(RPOT)) {
```

Comment:
Would it be sufficient to do a shallow traversal of the vector loop region?

Reply:
Looks like it, done in 4776c9c350e950ed1e276b5b2621343641058d4b